src/components/DonatePage/DonatePageBody.tsx

```diff
@@ -259,6 +254,15 @@ const DonatePageBody = () => {
     });
   };
 
+  const handleAcceptTerms = () => {
+    if (!user?.acceptedToS) {
+      setTerms(true);
+
+      // Save that user accepted terms and conditions
+      updateAcceptedTerms(true);
+    }
+  };
```

Comment:
In this way, you will send request to backend on every check/uncheck the user does if she hasn't accepted before.

Reply:
The best place in my view is before triggering transaction

Reply:
I putted on that way but they wanted to not be there, it should be saved on first terms accepted state :(